java/component/keyple-plugin/remote-se/nativese/src/main/java/org/eclipse/keyple/plugin/remotese/nativese/impl/TransmitExecutor.java

```diff
@@ -0,0 +1,97 @@
+import org.eclipse.keyple.core.seproxy.message.ProxyReader;
+import org.eclipse.keyple.core.seproxy.message.SeRequest;
+import org.eclipse.keyple.core.seproxy.message.SeResponse;
+import org.eclipse.keyple.core.util.json.KeypleJsonParser;
+import org.eclipse.keyple.plugin.remotese.core.KeypleMessageDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.gson.JsonObject;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * Execute locally a TRANSMIT KeypleMessageDto
+ */
+class TransmitExecutor implements Executor {
+
+    private static final Logger logger = LoggerFactory.getLogger(TransmitExecutor.class);
+
+    private final ProxyReader reader;
+
+    /**
+     * Builds a TRANSMIT KeypleMessageDto executor
+     */
+    TransmitExecutor(ProxyReader reader) {
+        this.reader = reader;
+    }
+
+    @Override
+    public KeypleMessageDto execute(KeypleMessageDto keypleMessageDto) {
+
+        KeypleMessageDto response = null;
+
+        // Extract info from keypleDto
+        JsonObject bodyObject =
+                KeypleJsonParser.getParser().fromJson(keypleMessageDto.getBody(), JsonObject.class);
+
+
+        ChannelControl channelControl =
+                ChannelControl.valueOf(bodyObject.get("channelControl").getAsString());
+
+        SeRequest seRequest = KeypleJsonParser.getParser()
+                .fromJson(bodyObject.get("seRequest").getAsString(), SeRequest.class);
+
+
+
+        if (logger.isTraceEnabled()) {
+            logger.trace("Execute locally seRequest : {} with params {} on reader {}", seRequest,
+                    channelControl, reader.getName());
+        }
+
+        try {
+
+            // execute transmit
+            SeResponse seResponse = reader.transmitSeRequest(seRequest, channelControl);
+
+            // prepare response body
+            response = new KeypleMessageDto(keypleMessageDto)
+                    .setBody(KeypleJsonParser.getParser().toJson(seResponse, SeResponse.class));;
+
+        } catch (KeypleReaderIOException e) {
+
+            JsonObject body = new JsonObject();
+            if (e.getSeResponses() != null) {
+                body.add("seResponses", KeypleJsonParser.getParser().toJsonTree(e.getSeResponses(),
+                        new TypeToken<List<SeResponse>>() {}.getType()));
+            }
+            if (e.getSeResponse() != null) {
+                body.add("seResponse", KeypleJsonParser.getParser().toJsonTree(e.getSeResponse(),
+                        SeResponse.class));
+            }
+
+            // if an exception occurs, send it into a keypleDto to the Master
+            response = new KeypleMessageDto(keypleMessageDto)
+                    .setAction(KeypleMessageDto.Action.ERROR.name())//
+                    .setErrorCode(KeypleMessageDto.ErrorCode.KeypleReaderIOException.getCode())//
+                    .setErrorMessage(e.getMessage())//
+                    .setBody(body.toString());
```

Comment:
Normalement les champs errorCode et body ne doivent pas être tous les deux settés. A discuter

Reply:
KeypleReaderIoException contains SeResponses, then we must send them to the server.